packages/hub/indexing/card-utils.ts

```diff
@@ -132,6 +153,19 @@ async function generateCardModule(card: SingleResourceDoc) {
       }
     }
     if (isEqual(cleanCard.data, cardOnDisk.data)) { return; }
+
+    try {
+      seenVersions = readJSONSync(join(cardFolder, seenVersionsFile));
+    } catch(e) {
+      if (e.code !== 'ENOENT') { throw e; } // ignore file not found errors
+    }
+    // the PendingChange class will actually create DocumentContext for the old
+    // and new versions of the document when it is being updated, and we dont want
+    // to inadvertantly clobber the latest card on disk with an older version of
+    // the card as a result of processing an older card as part of what PendingChange
+    // needs to do, so we keep track of the versions of the card that we've seen and
+    // only write to disk if the version of the card is not one we have encountered yet.
```

Comment:
I think the version should be part of the identity, at least in this particular area. In other words, the path on disk you're writing out includes the version, so that versions never clobber each other. The only extra state needed would be for optimization (like a special dotfile inside each card's folder that indicates we wrote it out successfully, so it's easy to test for that and skip doing it again).

Reply:
makes sense